src/main/java/org/terasology/servermotd/ServerMOTDSystem.java

```diff
@@ -16,30 +16,55 @@
 package org.terasology.servermotd;
 
 import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
-import org.terasology.network.NetworkMode;
-import org.terasology.network.NetworkSystem;
+import org.terasology.logic.console.commandSystem.annotations.Command;
+import org.terasology.logic.console.commandSystem.annotations.CommandParam;
+import org.terasology.logic.console.commandSystem.annotations.Sender;
 import org.terasology.registry.CoreRegistry;
-import org.terasology.registry.Share;
+import org.terasology.registry.In;
+
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ServerMOTDSystem extends BaseComponentSystem {
+    @In
+    private EntityManager entityManager;
 
-@Share(ServerMOTD.class)
-@RegisterSystem(RegisterMode.ALWAYS)
-public class ServerMOTDSystem extends BaseComponentSystem implements ServerMOTD {
     private Context context = CoreRegistry.get(Context.class);
 
-    @Override
+    private EntityRef MOTDEntity;
+    private MOTDProvider renderMOTD = new MOTDProvider();
```

Comment:
I wonder if you really need to leave the `MOTDProvider` as its own class. Couldn't you just add the convenience methods to the system instead?

I'm not totally sure if that's better or worse.

Reply:
I implemented the MOTDProvider as a separate class so that when the MOTD functionality from a file was implemented there would be a single place which would require all changes and also, all the code concerned with the procurement of the MOTD from whatever source was all in one place.